src/protocol/protocol.ml

```diff
@@ -3,10 +3,16 @@ module Location = Ocaml_parsing.Location
 
 type source = string
 
+type cmis = {
+  static_cmis : (string * string) list;
+  cmi_urls : string list
+}
+
 type action =
   | Complete_prefix of source * Msource.position
   | Type_enclosing of source * Msource.position
   | All_errors of source
+  | Init_cmis of cmis
```

Comment:
If we made cmi loading "on demand" I would prefer to call this `Add_cmis`.
This could also be used to provide user for completion with another libray of its choice.

Reply:
Right, I was thinking too much of the code from [js_top_worker](https://github.com/jonludlam/js_top_worker/blob/main/idl/toplevel_api.ml#L67-L80) where we not only have to setup these URLs but also have to call the module initialisation code.